pep-0690.rst

```diff
@@ -312,57 +316,73 @@ import object. When ``modb.foo`` is later referenced, it will also try to
+  flag set that indicates it might contain lazy objects, that flag will be
+  passed on to the updated/copied dictionary. This still ensures that the lazy
+  import object can't escape into Python code without being resolved.
+
+* Through the garbage collector: lazy imported objects are still Python objects
+  and live within the garbage collector; as such, they can be collected and seen
+  by means of using ``gc.collect()`` and ``gc.get_objects()``. Lazy objects are,
+  however, self contained and should pose no danger if extracted from the
+  garbage collector in this way.
+
+Other "bulk" dictionary lookup methods involving values (such as ``dict.items()``,
+``dict.values()``, ``PyDict_Next()`` etc.) will attempt to resolve *all* lazy
+import objects in the dictionary prior starting the lookup. Since only (some)
+module namespace dictionaries will ever have ``dk_lazy_imports`` set, the extra
+overhead of resolving all lazy import objects inside a dictionary is only paid
+by those dictionaries that need it. Minimizing the overhead on normal non-lazy
+dictionaries is the sole purpose of the ``dk_lazy_imports`` flag.
+
+Given the fact that ``PyDict_Next`` could attempt to resolve all lazy import
+objects the first time position ``0`` is being accessed, imported objects could
+fail with exceptions. In the cases where objects can't be resolved,
+``PyDict_Next`` will return ``0`` ignoring any errors.
+
+For this reason, this PEP introduces ``PyDict_NextWithError``, that works in the
+same way as ``PyDict_Next``, but which can set an error when returning ``0`` and
+this should be checked via ``PyErr_Ocurred()`` after the call.
 
 The eagerness of imports within ``try`` / ``except`` / ``with`` blocks or within
 class or function bodies is handled in the compiler via a new
 ``EAGER_IMPORT_NAME`` opcode that always imports eagerly. Top-level imports use
 ``IMPORT_NAME``, which may be lazy or eager depending on ``-L`` and/or
 ``importlib.set_lazy_imports()``.
 
+The current status of lazy imports at any given place is determined by a
+combination of the passed ``-L`` option flag; an interpreter-wide flag set by
+``importlib.set_lazy_imports()`` and the container object passed in its
+``excluding`` keyword argument; and a flag set by
+``importlib.enable_lazy_imports_in_module()`` in nearest running module frame.
+All these together are used to cache the current status of lazy imports in the
+currently running frame, this cache is globally busted whenever any of these API
+functions is called and, as such, changes are immediate for all new import
+statements.
 
-Exceptions
-----------
+
+Exceptions and Deferred Exceptions
+----------------------------------
 
 Exceptions that occur during a lazy import bubble up and erase the
 partially-constructed module(s) from ``sys.modules``, just as exceptions during
 normal import do.
 
 Since errors raised during a lazy import will occur later (wherever the imported
```

Comment:
I think the remainder of this probably belongs better below in the "drawbacks" section (or the "backwards compatibility" section where you removed it from) instead of here. (Or maybe it should be both places, with a quick specification of the behavior here and slightly more discussion of the backwards-compatibility and debugging implications below.)